sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/backend/rust/RustSqlRuntimeLibrary.java

```diff
@@ -180,6 +181,8 @@ public FunctionDescription getImplementation(
         if (opcode.equals(DBSPOpcode.DIV))
             // Always, for division by 0
             returnType = returnType.setMayBeNull(true);
+        if (opcode.equals(DBSPOpcode.SAFE_ADD))
```

Comment:
this should not be necessary hopefully, since the Calcite type checker should have done this already.
Calcite sometimes is very smart and may make this non-null if the inputs are constant and it has checked that there is no overflow.
What happens if you remove it?

Reply:
Is this supposed to overflow or not?
You can leave the mayBeNull there I guess.

Reply:
This is supposed to overflow.

Reply:
this is very strange, because the result is supposed to be NULL, so this looks like a Calcite bug...
Maybe this is a good enough workaround, the question is whether I spend time to investigate it in Calcite.